static/js/chart/base.test.ts

```diff
@@ -65,3 +66,43 @@ test("shouldFillInValues", () => {
   ];
   expect(shouldFillInValues(series)).toBe(true);
 });
+
+beforeEach(() => {
+  // JSDom does not define SVGTSpanElements, and use SVGElement instead. Defines
+  // a shim for getComputedTextLength where each character is 1 px wide.
+  (window.SVGElement as any).prototype.getComputedTextLength = function () {
```

Comment:
this is nice! I remember trying this last year but got no luck...

Reply:
yea, it was quite a project to get the tests going...